Added istio-csr feature section in cert-manager docs (#89608)
openshift/openshift-docs: modules/cert-manager-istio-csr-uninstalling.adoc

Reviewer: This seems fine, during / before uninstallation users would like to remove the istio-csr stuff from their cluster.
However, you can check if this section is more suitable to be added to the new page I suggested, somewhere like in below section "Cleanup .." or the alike.

+
[source,terminal]
----
$ oc -n <istio_csr_project_name> delete istiocsrs.operator.openshift.io default
----

. Remove related resources by running the following commands
+
[IMPORTANT]
====
To avoid disrupting any {SMProductName} or Istio components, ensure that no component is referencing the Istio-CSR service or the certificates issued for Istio before removing the following resources.
====

.. Delete the cluster scoped resources listed by running the following command:
+
[source,terminal]
----
$ oc get clusterrolebindings.rbac.authorization.k8s.io,clusterroles.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr"
----

.. Delete the resources in Istio-csr deployed namespace, listed by running the following command:
+
[source,terminal]
----
$ oc get certificate.cert-manager.io,deployments.apps,services,serviceaccounts -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
----

.. Delete the resources in {SMProductName} or Istio deployed namespace listed by running the following command:
+
[source,terminal]
----

> $ oc get roles.rbac.authorization.k8s.io,rolebindings.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>

Reviewer 2: @swghosh @bharath-b-rh @lunarwhite Is this technically correct? Do these commands delete the resources as described in the steps?

Reviewer 3: @snarayan-redhat The commands are for listing the resources, and user has to delete the listed resources.

Reviewer 2: That was my understanding as well. But the wordings doesn't seem to imply it well. 

@subhtk `Delete..by running the following command`. This command is for listing and not deleting. Deletion would be the following action. It must be something like `Fetch the list of resources...` e.g.
```
.. Delete the cluster scoped resources listed by running the following command:
+
[source,terminal]
----
$ oc get clusterrolebindings.rbac.authorization.k8s.io,clusterroles.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr"
----
```
@skopacz1 Maybe you can help @subhtk fix the wordings?

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-operator-integrating-istio.adoc

:_mod-docs-content-type: PROCEDURE
[id="cert-manager-istio-csr-uninstalling_{context}"]

> = Uninstalling the {cert-manager-operator} with istio-csr 

Reviewer 4: ```suggestion
= Uninstalling the Istio-CSR agent managed by {cert-manager-operator}
```

> To uninstall the {cert-manager-operator} with Istio-CSR, use the following steps.

Reviewer 4: ```suggestion
Use this procedure to uninstall the Istio-CSR agent managed by {cert-manager-operator}.
```

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.

> * You have enabled the `Istiocsr`.

Author: It is not duplicated. What I mean here is: "You have enabled the `Istiocsr` feature."

> * You have enabled the Istio-CSR.
> * You have created the `IstioCSR` custom resource.

Reviewer 4: ```suggestion
* You have installed the Istio-CSR agent through {cert-manager-operator}.
```

> .Procedure

Reviewer 5: This procedure is still missing the first step that was here earlier

Reviewer 4: +1 (same as https://github.com/openshift/openshift-docs/pull/89608#discussion_r1994079425

Author: Added

> 

Reviewer 4: Missing step one:

1. Remove the `IstioCSR` custom resource by running the following command:

```
$ oc -n <istio-csr_project_name> delete istiocsr.operator.openshift.io default
```

. Remove related resources:
+
[IMPORTANT]
====
To avoid disrupting any {SMProductName} or Istio components, ensure that no component is referencing the Istio-CSR service or the certificates issued for Istio before removing the following resources.
====

.. List the cluster scoped-resources by running the following command and save the names of the listed resources for later reference:
+
[source,terminal]
----

> $ oc get clusterrolebindings.rbac.authorization.k8s.io,clusterroles.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr"

Reviewer 4: ```suggestion
$ oc get clusterrolebindings,clusterroles -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr"
```

----

.. List the resources in Istio-csr deployed namespace by running the following command and save the names of the listed resources for later reference:
+
[source,terminal]
----

> $ oc get certificate.cert-manager.io,deployments.apps,services,serviceaccounts -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>

Reviewer 4: ```suggestion
$ oc get certificate,deployments,services,serviceaccounts -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
```

----

.. List the resources in {SMProductName} or Istio deployed namespaces by running the following command and save the names of the listed resources for later reference:
+
[source,terminal]
----

> $ oc get roles.rbac.authorization.k8s.io,rolebindings.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>

Reviewer 4: ```suggestion
$ oc get roles,rolebindings -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
```

----

.. Create the `Deployment` resources by running the following command:
+
[source,terminal]
----
$ oc get deployments.apps -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
----

.. Create the `Roles` resources by running the following command:
+
[source,terminal]
----
$ oc get roles.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
----

.. Create the `RoleBinding` resources by running the following command:
+
[source,terminal]
----
$ oc get rolebindings.rbac.authorization.k8s.io -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
----

.. Create the `Service` resources by running the following command:
+
[source,terminal]
----
$ oc get services -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio_csr_project_name>
----

.. Create the `ServiceAccount` resources by running the following command:
+
[source,terminal]
----
$ oc get serviceaccounts -l "app=cert-manager-istio-csr,app.kubernetes.io/name=cert-manager-istio-csr" -n <istio-csr_project_name>

> ----

Author: Done
